### PR TITLE
Add the error of validateExtensionProvider to errors when validate MeshConfig.

### PR DESCRIPTION
**Pull request by devincd**
Signed-off-by: devincd <505259926@qq.com>

**Please provide a description of this PR:**
I think it should add the error of validateExtensionProvider to the errors that validateMeshConfig returns. If some extensionProviders are invalid, there are can not be ignore, because we do not know which extension provider is used or not used.

Related issue: https://github.com/istio/istio/issues/34693

### CODE REVIEW

**Comment by devincd** [comment]
@hzxuzhonghu Please take a look, thinks.

**Comment by devincd** [comment]
@incfly Please take a look, thinks.

**Comment by bianpengyuan** [comment]
This reverts https://github.com/istio/istio/pull/30878 cc @yangminzhu

**Comment by incfly** [comment]
@yangminzhu any context on #30824 ? what issue we are trying to address there? if the default install succeed, the error only happens when human edit meshconfig occurs right? by that time, they could fix the invalid config via intervention.

**Comment by yangminzhu** [comment]
> @yangminzhu any context on #30824 ? what issue we are trying to address there? if the default install succeed, the error only happens when human edit meshconfig occurs right? by that time, they could fix the invalid config via intervention.

This is to provide a smoother user experience, the user might be trying out multiple configs and some of them could be invalid transiently, we don't want to force the user to delete them unless the config is actually being used.

**Review by yangminzhu — CHANGES_REQUESTED** [review]
can we instead fix this directly in pilot/pkg/security/authz/builder/extauthz.go? We intentionally didn't want to block the user from a config that might not be actively used in the MeshConfig.

**Comment by devincd** [comment]
> can we instead fix this directly in pilot/pkg/security/authz/builder/extauthz.go? We intentionally didn't want to block the user from a config that might not be actively used in the MeshConfig.

I try to fix.

**Comment by devincd** [comment]
@yangminzhu Please take a look, thinks

File: pilot/pkg/security/authz/builder/extauthz.go [hunk 1]
```diff
@@ -84,9 +85,19 @@ func processExtensionProvider(in *plugin.InputParams) map[string]*builtExtAuthz
...
-			parsed, err = buildExtAuthzHTTP(in, p.EnvoyExtAuthzHttp)
+			errValidation := validation.ValidateExtensionProviderEnvoyExtAuthzHTTP(p.EnvoyExtAuthzHttp)
+			if errValidation != nil {
+				errs = multierror.Append(errs, errValidation)
+			} else {
+				parsed, err = buildExtAuthzHTTP(in, p.EnvoyExtAuthzHttp)
+			}
 		case *meshconfig.MeshConfig_ExtensionProvider_EnvoyExtAuthzGrpc:
-			parsed, err = buildExtAuthzGRPC(in, p.EnvoyExtAuthzGrpc)
+			errValidation := validation.ValidateExtensionProviderEnvoyExtAuthzGRPC(p.EnvoyExtAuthzGrpc)
```
[the remark this answers is not in the document]
Comment (devincd) [reply pilot/pkg/security/authz/builder/extauthz.go:95]: yes, this is more better.

**Review by yangminzhu — APPROVED** [review]
thanks!

**Comment by devincd** [comment]
@hzxuzhonghu PTAL,thinks.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**